fiftyone/core/metadata.py

```diff
@@ -382,28 +389,79 @@ def compute_metadata(
             default value ``fiftyone.config.show_progress_bars`` (None), or a
             progress callback function to invoke instead
     """
-    num_workers = fou.recommend_thread_pool_workers(num_workers)
+    logger.info("Computing metadata...")
 
     if sample_collection.media_type == fom.GROUP:
         sample_collection = sample_collection.select_group_slices(
             _allow_mixed=True
         )
 
-    if num_workers <= 1:
-        _compute_metadata(
-            sample_collection, overwrite=overwrite, progress=progress
-        )
-    else:
-        _compute_metadata_multi(
+    if not overwrite:
+        sample_collection = sample_collection.exists("metadata", False)
+
+    if (sample_count := len(sample_collection)) == 0:
+        logger.info("No samples to compute metadata for")
+        return
+
+    if num_workers == 1:
+        return _compute_metadata_single(
             sample_collection,
-            num_workers,
             overwrite=overwrite,
             progress=progress,
+            num_samples=sample_count,
+            skip_failures=skip_failures,
+            warn_failures=warn_failures,
         )
 
+    if not num_workers:
+        num_workers = fou.recommend_thread_pool_workers(num_workers)
+
+    mapper = focm.MapperFactory.create(
+        "process",
```

Comment:
We added a `default_parallelization_method` config setting specifically so that users could globally configure whether FO should processes or threads to parallelize things in the SDK.

So directly using `"process"` here is a big no-no! We should leverage the core abstractions we provide like `SampleCollection.map_samples()` to get work done and not directly invoke specific backends.

Reply:
ok I kept it from josey's original implementation and it seemed that process was always faster, but if there is already a config, I can just use that.